mobile/components/views/Attractions/Search/Search.tsx

```diff
@@ -1,122 +1,110 @@
-    category: "",
-    city: "",
-    country: "",
-    keyword: "",
-    radius: "10",
-    region: ""
+    category: '',
+    city: '',
+    country: '',
+    keyword: '',
+    radius: '10',
+    region: '',
   });
 
-  const [radiuses, setRadiuses] = useState<string[]>([]);
-  const navigation = useNavigation<NavigationProps>();
-
+  const [radii, setRadii] = useState<string[]>([]);
 
   useEffect(() => {
     const radiusData = ['5', '10', '15'];
-    setRadiuses(radiusData);
-    fetchUsersAttractions(1)
-      .then((attractions) => setUserAttractionsData(attractions))
-      .catch((error) => {
-        if (userAttractionsData === attractionsExamples) {
+    setRadii(radiusData);
+    fetchUserAttractions(1)
+      .then((fetchAttractions) => setUserAttractions(fetchAttractions))
+      .catch(() => {
+        if (userAttractions === attractionsExamples) {
           Alert.alert('Error', 'Failed getting user attractions. Loading demo data');
         } else {
           Alert.alert('Error', 'Failed getting user attractions.');
         }
       })
       .then(() => setLoading(false));
-  }, []);
-
+  }, [userAttractions]);
 
   useEffect(() => {
-    const filteredAttractions = userAttractionsData.filter(attraction => {
+    const filteredAttractions = userAttractions.filter((attraction) => {
       return (
-        (filters.category === "" || attraction.category === filters.category) &&
-        (filters.city === "" || attraction.city === filters.city) &&
-        (filters.country === "" || attraction.country === filters.country) &&
-        (filters.keyword === "" || attraction.name.toLowerCase().includes(filters.keyword.toLowerCase())) &&
-        (filters.region === "" || attraction.region.toLowerCase().includes(filters.region.toLowerCase()))
+        (filters.category === '' || attraction.category === filters.category) &&
+        (filters.city === '' || attraction.city === filters.city) &&
+        (filters.country === '' || attraction.country === filters.country) &&
+        (filters.keyword === '' ||
+          attraction.name.toLowerCase().includes(filters.keyword.toLowerCase())) &&
+        (filters.region === '' ||
+          attraction.region.toLowerCase().includes(filters.region.toLowerCase()))
       );
     });
     setAttractions(filteredAttractions);
-  }, [filters]);
+  }, [filters, userAttractions]);
 
-  const handleSetFilters = (filtersProps: Filters) => {
+  const onSetFilters = (filtersProps: Filters) => {
     setFilterModalVisible(false);
     setFilters(filtersProps);
   };
 
-  const handleAttractionClick = (attraction: Attraction) => {
-    // setAttractionModalVisible(true);
-    setAttractionDetails(attraction);
-    navigation.navigate('AttractionDetailScreen', { id: attraction.id });
-
+  const onAttractionClick = (attraction: Attraction) => {
+    let attractionId = attraction?.id;
+    navigation.navigate('AttractionDetailScreen', { id: attractionId });
```

Comment:
For me there is no need to creat let attraction variable, it will be okay if you will use in navicate `{id: attraction?.id}`

Reply:
Here, omitting a variable resulted in a warning. I don't like it either, but it's necessary.